force-app/main/default/classes/SOQL.cls

```diff
@@ -258,8 +362,18 @@ public with sharing class SOQL {
      * @param accountId The ID of the Account.
      * @return An Account record with related Opportunities.
      */
+
     public static Account getAccountAndOpportunitiesFromAccountId(Id accountId) {
-        return null; // Replace null with the variable you used to store the result
+        List<Account> accountListWithOpps = new List<Account>([
+            SELECT Id, Name, (SELECT Id, Name, StageName, CloseDate, AccountId FROM Opportunities)
+            FROM Account
+            WHERE Id =:accountId WITH SECURITY_ENFORCED
+        ]);
+
+        if (!accountListWithOpps.isEmpty()) {
+            return accountListWithOpps[0];
+        }
+        return accountListWithOpps[1]; 
```

Comment:
If accountListWithOpps is empty on line 373 then the code will reach this line, but there won't be anything at the 1 index to return. It would make more sense to return null here to avoid any errors/exceptions being thrown